packages/core/test/insured-bridge/e2e/helpers/OptimismConstants.js

```diff
@@ -1,4 +1,10 @@
 // Pre-loaded wallet from Optimism docker containers with 10k eth on both L1 and L2.
 const DEFAULT_ADMIN_KEY = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
 
-module.exports = { DEFAULT_ADMIN_KEY };
+// To correctly deploy on optimism, some settings need to be set.
+const OPTIMISM_GAS_OPTS = { gasLimit: 12_500_000, gasPrice: 0 };
+
+// Address of the proxied cross-domain messenger deployed in the default Optimism container. Not exported by the package.
+const PROXY__OVM_L1_CROSS_DOMAIN_MESSENGER = "0x59b670e9fa9d0a427751af201d676719a970857b";
```

Comment:
getting this address right was a bit of a pain. it needs to be set like this to work correctly. It also means that we cant use the `Lib_PredeployAddresses` in the withdraw box.

Reply:
Curious where you got this address from? And do you think we need to hardcode this address in the `BridgePoolFactory` as the `_crossDomainMessenger`?

Reply:
I got it from one of the optimism example tests. Also, if you are running the optimism docker containers and go to http://localhost:8080/addresses.json you will see this:
```
{
  "AddressManager": "0x5FbDB2315678afecb367f032d93F642f64180aa3",
  "OVM_CanonicalTransactionChain": "0xA51c1fc2f0D1a1b8494Ed1FE312d7C3a78Ed91C0",
  "OVM_ChainStorageContainer-CTC-batches": "0x0165878A594ca255338adfa4d48449f69242Eb8F",
  "OVM_ChainStorageContainer-CTC-queue": "0x2279B7A0a67DB372996a5FaB50D91eAA73d2eBe6",
  "OVM_ChainStorageContainer-SCC-batches": "0x610178dA211FEF7D417bC0e6FeD39F05609AD788",
  "OVM_ExecutionManager": "0xa85233C63b9Ee964Add6F2cffe00Fd84eb32338f",
  "OVM_FraudVerifier": "0x7a2088a1bFc9d81c55368AE168C2C02570cB814F",
  "OVM_L1CrossDomainMessenger": "0x68B1D87F95878fE05B998F19b66F4baba5De1aed",
  "OVM_L1MultiMessageRelayer": "0xa82fF9aFd8f496c3d6ac40E2a0F282E47488CFc9",
  "OVM_SafetyChecker": "0x84eA74d481Ee0A5332c457a4d796187F6Ba67fEB",
  "OVM_StateCommitmentChain": "0x9A676e781A523b5d0C0e43731313A708CB607508",
  "OVM_StateManagerFactory": "0xc5a5C42992dECbae36851359345FE25997F5C42d",
  "OVM_StateTransitionerFactory": "0xE6E340D132b5f46d1e472DebcD681B2aBc16e57E",
  "Proxy__OVM_L1CrossDomainMessenger": "0x59b670e9fA9D0A427751Af201D676719a970857b",
  "Proxy__OVM_L1StandardBridge": "0x851356ae760d987E095750cCeb3bC6014560891C",
  "OVM_BondManager": "0x959922bE3CAee4b8Cd9a407cc3ac1C251C2007B1",
  "OVM_Sequencer": "0x70997970C51812dc3A010C7d01b50e0d17dc79C8",
  "Deployer": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
}
```

in here you can find that address!

Reply:
also we dont wnat to hardcode it in the `BridgePoolFactory` as it will be different on mainnet. On mainnet it's here: https://etherscan.io/address/0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1

you can find the details of this and other contracts here: https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts/deployments/README.md